renderer/src/components/WalletModule.tsx

```diff
@@ -0,0 +1,106 @@
+import { FC, useEffect, useState } from 'react'
+import HeaderBackgroundImage from '../assets/img/header-curtain.png'
+import { ReactComponent as EditIcon } from '../assets/img/icons/edit.svg'
+
+import FilAddressForm from './FilAddressForm'
+import WalletTransactionsHistory from './WalletTransactionsHistory'
+import useWallet from '../hooks/StationWallet'
+import TransferFundsButtons from './TransferFunds'
+import { trasnferAllFundsToDestinationWallet } from '../lib/station-config'
+
+interface PropsWallet {
+  isOpen: boolean,
+}
+
+const WalletModule: FC<PropsWallet> = ({ isOpen = false }) => {
+  const [editMode, setEditMode] = useState<boolean>(false)
+  const [trasnferMode, setTransferMode] = useState<boolean>(false)
+  const [stationAddress, destinationAddress, walletBalance, allTransactions, setDestinationAddress, latestTransaction, dismissLatestTransaction] = useWallet()
+
+  useEffect(() => {
+    dismissLatestTransaction()
+    reset()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen])
+
+  const reset = () => {
+    setEditMode(false)
+    setTransferMode(false)
+  }
+
+  const enableEditMode = () => {
+    setEditMode(true)
+    setTransferMode(false)
+  }
+
+  const enableTransferMode = () => {
+    setEditMode(false)
+    setTransferMode(true)
+  }
+
+  const saveAddress = async (address: string | undefined) => {
+    setDestinationAddress(address)
+    setEditMode(false)
+  }
+
+  const transferAllFunds = async () => {
+    await trasnferAllFundsToDestinationWallet()
+    setTransferMode(false)
+  }
+
+  return (
+    <div className='relative'>
+      <div className='h-8 bg-primary-dark flex items-center px-8' onClick={reset}>
+        <p className='text text-body-3xs text-white opacity-80 mr-3'>STATION ADDRESS</p>
+        <p className='text text-body-3xs text-white'>{stationAddress}</p>
+      </div>
+      <div className='h-60 bg-primary bg-no-repeat bg-center' style={{ backgroundImage: `url(${HeaderBackgroundImage})` }}>
+        <div className="py-6 px-6 reset">
+          <div className="flex flex-row justify-between align-baseline">
+            {editMode || !destinationAddress
+              ? <div className='w-full' onClick={() => (setEditMode(true))}>
+                <FilAddressForm
+                  destinationAddress={destinationAddress}
+                  saveDestinationAddress={saveAddress}
+                  editMode={editMode}/>
+                </div>
+              : <div className="w-full flex flex-col z-0 items-start mb-[31px]" onClick={() => { setTransferMode(false) }}>
```

Comment:
These ternary expressions are a bit noisy, what about extracting the markup into `render` functions? IE: `renderAddressForm()` ?